src/printer.rs

```diff
@@ -305,6 +306,9 @@ impl<W: WriteColor> Printer<W> {
             self.write_path_sep(b':');
         }
         if let Some(line_number) = line_number {
+            if line_number > buf.lines().count() as u64 {
+                return;
+            }
```

Comment:
Sorry, but this just won't work on multiple levels. Firstly, `buf` doesn't necessarily contain the entire contents of the file being searched (this *is* true for the memory map searcher only). Secondly, `buf.lines().count()` is recounting all of the lines in the buffer **every time a match is written**, which would be a devastating performance regression.

Reply:
Third, if the end user disabled line numbers, then this check won't run at all.